partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/FnPlus.kt

```diff
@@ -16,33 +19,53 @@ internal object FnPlus : DiadicArithmeticOperator("plus") {
 
     override fun getTinyIntInstance(tinyIntLhs: PType, tinyIntRhs: PType): Function.Instance {
         return basic(PType.tinyint()) { args ->
-            @Suppress("DEPRECATION") val arg0 = args[0].byte
-            @Suppress("DEPRECATION") val arg1 = args[1].byte
-            Datum.tinyint((arg0 + arg1).toByte())
+            val arg0 = args[0].byte
+            val arg1 = args[1].byte
+            val result = arg0 + arg1
+            if (result.byteOverflows()) {
+                throw DataException("Numeric value out of range for TINYINT: $arg0 + $arg1")
```

Comment:
```suggestion
                throw DataException("Resulting value out of range for TINYINT: $arg0 + $arg1")
```

Non-blocking, but the use of "numeric" might be confusing due to there being a numeric data type.